modules/ROOT/pages/lb-mapping-rules.adoc

```diff
@@ -8,165 +8,278 @@ image:logo-hybrid-disabled.png[link="/runtime-manager/deployment-strategies", ti
+----
 
-The external load balancer domain name depends on the unique name you assign to it. In the following mapping rule examples, the load balancer is `lb-demo`.
+To determine the load balancer ID, perform a `GET` request to your endpoint:
 
-By default, your load balancer listens to external requests on HTTPS and communicates internally with your worker over HTTP. If you configured your Mule application within the VPC to listen on HTTPS, make sure you set `upstreamProtocol` to `HTTPS` when creating the mapping list using the xref:anypoint-platform-cli.adoc#cloudhub-load-balancer-mappings-add[`load-balancer mappings add`] command.
+[source,text,linenums]
+----
+/organizations/{orgid}/loadbalancers
+----
 
-=== URL Mapping
 
-You can pass the app name as an input URI and map it directly to the app name in CloudHub:
+== Using a Literal 1:1 Mapping
+
+If you have only one application, you can map the literal app name.
+In this case, the rule must match the application exactly. 
+
+This rule maps the default load balancer `lb-demo.lb.anypointdns.net` directly to the `myApp.cloudhub.io` app in CloudHub:
+
+[%header,cols="10a,20a,20a,20a,10a"]
+|===
+| Rule # | *Input Path* | *Target App* | *Output Path* | *Protocol*
+| 1 | / | myApp | / | http 
+|===
+
+This rule matches the `dev-misc-mythellotest-v1-myapp` app:
 
-[%header,cols="10a,20a,20a,10a"]
+[%header,cols="10a,20a,20a,20a,10a"]
 |===
-|Rule # |Input URL 2+^| Output URL
-|   |    *URI*   |       *appName*   |   *appURI*
-| 0 | /{app}/    | {app}             | /
+| Rule # | *Input Path* | *Target App* | *Output Path* | *Protocol*
+| 1 | /{layera}/{domaina}/v{versiona}/ | dev-{layera}-{domaina}-{versiona}-myapp | /api/ | http
 |===
 
-This rule maps `lb-demo.lb.anypointdns.net/{app}` to `{app}.cloudhub.io`. `{app}` is a pattern for any application name you pass in the inbound request.
 
+== Using Patterns in Mapping Rules
 
-=== Host Mapping
+A _pattern_ is a string that defines a template for matching an input text. The value in curly brackets (`{   }`) is treated as a variable.
 
-If you have a wildcard certificate (such as `*.example.com`), you can use the `subdomain` variable to map any subdomain:
+* Variable *names* can contain only lowercase letters (`a-z`) and no other characters.
+* Variable *values* can contain the characters `a-z0-1.&?-_`, but can't contain slash characters (`/`).
 
-[%header,cols="10a,20a,20a,10a"]
+
+[IMPORTANT]
+DLBs don't support patterns in the *Output Path* (`appURI`).
+
+If you set up a DNS CNAME record to map `example.com` to `_lb-name_.lb.anypointdns.net`,
+you can then use a mapping rule to map `app.example.com` to a different deployed CloudHub Mule application name.
+
+[%header,cols="10a,20a,20a,20a,10a"]
 |===
-|Rule # |Input URL 2+^| Output URL
-|   |  *URI*   |       *appName*    |  *appURI*
-| 0 | /        | {subdomain}   | /
+| Rule # | *Input Path* | *Target App* | *Output Path* | *Protocol*
+|1 | /api/v1/ | default-app | /api/v1/ | http
```

Comment:
Thiss is going to catch traffic to second-app, too. It'll pass it to default-app/api/v1/second-app/<uri-params>

Reply:
I got this example from this KB article: https://help.mulesoft.com/s/article/Check-list-of-using-DLB-mapping-rules
> Take a real case for an example, with the rules below, URL "/api/v1/secondapp" will be mapped to the app "default-app" rather "second-app". In order to pick the more exact mapping, you need to put the second rule on top of the first.

Reply:
> 3. Pay attention to the sequence of the rules. DLB always picks up the first matching rule regardless if there are more exact matchings later. Take a real case for an example, with the rules below, URL "/api/v1/secondapp" will be mapped to the app "default-app" rather "second-app". In order to pick the more exact mapping, you need to put the second rule on top of the first.

It's the "bad" example

Reply:
Yes, I added another table:

 In the following example, the `/api/v1/secondapp` URL maps to the `default-app` app:

[%header,cols="10a,20a,20a,20a,10a"]
|===
|Priority|Input Path	|Target App| Output Path| Protocol
|1 | /api/v1/ | default-app | /api/v1/ | http
|2 | /api/v1/secondapp | second-app | /api/v1 |	http
|===

For the rule to match only `second-app`, the rule priority must be:

[%header,cols="10a,20a,20a,20a,10a"]
|===
|Priority|Input Path	|Target App| Output Path| Protocol
|1 | /api/v1/secondapp | second-app | /api/v1 |	http
|2 | /api/v1/ | default-app | /api/v1/ | http
|===

Reply:
Looks better, just get /api/v1/secondapp/ with that trailing slash

Reply:
Added.